src/hooks/users/useWorldUsers.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { createContext, useContext, useMemo } from "react";
+import { isEqual } from "lodash";
+
+import { User } from "types/User";
+
+import { WithId } from "utils/id";
+import { worldUsersWithoutLocationSelector } from "utils/selectors";
+import { isDefined } from "utils/types";
+
+import { isLoaded, useFirestoreConnect } from "hooks/useFirestoreConnect";
+import { useSelector } from "hooks/useSelector";
+import { useSovereignVenueId } from "hooks/useSovereignVenueId";
+
+const noUsers: WithId<User>[] = [];
+
+export interface WorldUsersContextState {
+  isSovereignVenueIdLoading: boolean;
+  sovereignVenueId?: string;
+  sovereignVenueIdError?: string;
+
+  shouldConnect: boolean;
+}
+
+const WorldUsersContext = createContext<WorldUsersContextState | undefined>(
+  undefined
+);
+
+export interface WorldUsersProviderProps {
+  venueId?: string;
+}
+
+export const WorldUsersProvider: React.FC<WorldUsersProviderProps> = ({
+  venueId,
+  children,
+}) => {
```

Comment:
I don't understnad what do we gain by having this in the Context. It basically does what `useConnectUsers` did with no additions?

Reply:
It also provides us explicit guarantees that the data has actually been connected, because it will throw errors otherwise.

This will also likely become more relevant in future followup PRs where I rip out `react-redux-firebase` and connect using the firebase SDK directly.

Reply:
Explained this a bit more in https://github.com/sparkletown/sparkle/pull/1703#issuecomment-876312854